### PR TITLE
Time sleep optimization on tests.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Just now we are waiting second to be ensure that the new created object has a later creation time. To be more accurate we need to wait difference between next second after last creation time of object and current time. 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/util/util.go [hunk 1]
```diff
@@ -729,3 +729,10 @@ func NewTestingLogger(writer io.Writer, level int) logr.Logger {
 		zap.Level(zapcore.Level(level)),
 		opts)
 }
+
+// WaitForNextSecondAfterCreation wait time between the start of the next second
+// after creationTimestamp and the current time to ensure that the new
+// created object has a later creation time.
+func WaitForNextSecondAfterCreation(obj client.Object) {
+	time.Sleep(time.Until(obj.GetCreationTimestamp().Add(time.Second).Truncate(time.Second)))
```
Comment (trasc) [inline test/util/util.go:737]: I don't think we need to Truncate.

Comment (mbobrovskyi) [reply test/util/util.go]: Yeap, you right. Thanks!

**Review by trasc — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**